src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1434,6 +1444,69 @@ public int executeHashNotCalculated(Rope rope) {
         }
     }
 
+    @NodeChildren({
+            @NodeChild(type = RubyNode.class, value = "encoding"),
+            @NodeChild(type = RubyNode.class, value = "bytes"),
+            @NodeChild(type = RubyNode.class, value = "start"),
+            @NodeChild(type = RubyNode.class, value = "end")
+    })
+    public abstract static class EncodingLengthNode extends RubyNode {
```

Comment:
Is this to compute the length of a single character or the character length between `start` and `end`?
I think the name could be clearer: `CharacterLengthNode` or so (same for the Precise version).

Reply:
I like that this basically matches the MRI name for the function. Here, it's based off the name of the function from `StringSupport` it's replacing. I could be open to a rename, but I'd rather save that for another PR then.

Reply:
I think some names in MRI/jcodings/StringSupport are unfortunately not very good.
Clarity is much more important and adding a comment about how it's named in MRI seems fine to me.
"the length of an encoding" makes no sense to me.
I think it'd be nice to directly start with a well-named node rather than rename later.

Is this counting multiple characters or it only counts one?

Reply:
I really don't want to rename it. I think it creates more problems than it solves because now I need to keep doing some mental remapping of names.

Per the javadoc, this is what `Encoding#length` does:

```
/**
     * Returns character length given stream, character position and stream end
     * returns <code>1</code> for singlebyte encodings or performs sanity validations for multibyte ones
     * and returns the character length, missing characters in the stream otherwise
     *
     * @return
     *  0               Never
     *  &gt; 0             Valid character, length returned
     *  -1              Illegal/malformed character
     *  &lt; -1 (-1 - n)   Number of missing bytes for character in p...end range
     *
     * Oniguruma equivalent: <code>mbc_enc_len</code>
     * modified for 1.9 purposes,
     */
```

Reply:
OK.
That JavaDoc is very hard to read.
So it returns the length in bytes of the character starting at index `p`, right?

Reply:
I believe so. Yes.